### ISSUE TITLE
History suggestion and autocompletion broken after chezmoi prompts

### ISSUE THREAD

**Issue by Nagidal** [nushell/nushell#8344]
### Describe the bug

I use [chezmoi](https://www.chezmoi.io/) to manage my dotfiles. Chezmoi sometimes has these interactive prompts about skipping or overwriting files when syncing my dotfiles. They work well in both PowerShell and Nushell

![autocompletion_after_chezmoi_prompts_in_powershell](https://user-images.githubusercontent.com/7075397/223352421-7c8e1124-d80a-4cfe-87d7-58bbfb96dd3b.gif)
As you can see, after the `chezmoi update` process finishes, I can still use the autocomplete feature of PowerShell.

In Nushell, autocompletion and accepting suggestions from the command history work well, as you can see at the beginning of this other animation. But after the `chezmoi update` procedure with the interactive prompts, the attempt to autocomplete or accept a suggestion from the Nushell command history completely scrambles the prompt, look:

![autocompletion_after_chezmoi_prompts_in_nushell](https://user-images.githubusercontent.com/7075397/223353915-fab482fe-d8a4-4563-896e-ddb8bd3ae38e.gif)

Once it is scrambled like that I cannot even exit Nushell. I have to kill the terminal window. (This happens with Nushell running in the ConEmu64 Terminal, Nushell in Alacritty, and also in other PowerShell.exe, so I think that this is a Nushell issue)

### How to reproduce

1. run a `chezmoi` command which has interactive prompts
2. then type something which triggers a suggestion from Nushell history or try to autocomplete something

### Expected behavior

autocompletion and suggestions from command history work as expected after the interactive chezmoi prompt.

### Screenshots

see the embedded gifs

### Configuration

| key                | value                                       |
| ------------------ | ------------------------------------------- |
| version            | 0.76.0                                      |
| branch             |                                             |
| commit_hash        |                                             |
| build_os           | windows-x86_64                              |
| build_target       | x86_64-pc-windows-msvc                      |
| rust_version       | rustc 1.69.0-nightly (44cfafe2f 2023-03-03) |
| rust_channel       | nightly-x86_64-pc-windows-msvc              |
| cargo_version      | cargo 1.69.0-nightly (9880b408a 2023-02-28) |
| pkg_version        | 0.76.0                                      |
| build_time         | 2023-03-06 13:38:57 +01:00                  |
| build_rust_channel | release                                     |
| features           | database, default, trash, which, zip        |
| installed_plugins  |                                             |

**Comment by fdncred** [comment]
To me, this looks something like the crossterm raw state isn't getting turned off. That may be happening in reedline or nushell... I think.

**Comment by Nagidal** [comment]
Here is a trace level log of how the suggestions and autocompletions first work, then `chezmoi update` runs, and then the error is there.

[file.zip](https://github.com/nushell/nushell/files/10910538/file.zip)

**Comment by fdncred** [comment]
I don't believe `failed to get console mode for stdout: The handle is invalid.` is coming from nushell

**Comment by Nagidal** [comment]
> I don't believe `failed to get console mode for stdout: The handle is invalid.` is coming from nushell

Could it be that it's coming from chezmoi then? But in PowerShell this error does not happen. Maybe @twpayne from chezmoi can tell what's going on here.

**Comment by fdncred** [comment]
@Nagidal I didn't change much but I'd be interested to see if #8373 makes any difference in your scenario. Can you test it please and report back?

**Comment by Nagidal** [comment]
> @Nagidal I didn't change much but I'd be interested to see if #8373 makes any difference in your scenario. Can you test it please and report back?

Alas, same error: `failed to get console mode for stdout: The handle is invalid.` on line 7652 of this new trace file
[nulog.log](https://github.com/nushell/nushell/files/10939356/nulog.log)

| key                | value                                    |
| ------------------ | ---------------------------------------- |
| version            | 0.76.1                                   |
| branch             | add_error_checking_to_vt                 |
| commit_hash        | 21943bdb9965e5ca67a3881e6ddcf5cd27793030 |
| build_os           | windows-x86_64                           |
| build_target       | x86_64-pc-windows-msvc                   |
| rust_version       | rustc 1.66.1 (90743e729 2023-01-10)      |
| rust_channel       | 1.66.1-x86_64-pc-windows-msvc            |
| cargo_version      | cargo 1.66.1 (ad779e08b 2023-01-10)      |
| build_time         | 2023-03-10 07:21:16 +01:00               |
| build_rust_channel | debug                                    |
| features           | default, zip                             |
| installed_plugins  |                                          |
